src/service/investment_date_report/investment_date_report.py

```diff
@@ -70,43 +71,65 @@ def get_by_metric_records(
                     sizes,
                     self.company_anonymization.companies,
                 )
-                data[id]["metrics"] = self.get_valid_year_records(company, years)
+                metric_values = self.get_records_by_invest_year(company, companies)
+                data[id]["metrics"] = metric_values
+                metric_data.update(self.sort_metric_values(metric_values))
+                data[id]["metrics"].update(metric_data)
+                companies_data[id] = company
 
-                metric_data.update(self.sort_metric_values(data[id]["metrics"]))
-                company_data["id"] = company["id"]
-                company_data["name"] = company["name"]
+        return companies_data
 
-        return [company_data, metric_data]
+    def get_records_by_invest_year(self, company: dict, companies_by_invest) -> dict:
+        years = [companies_by_invest[company["id"]]]
+        years = self.repository.get_years(years)
+        metric_data = self.get_valid_year_records(company, years)
 
-    def get_by_company_metrics(
-        self, metrics: list, years: list, access: bool, company_id, **conditions
+        return metric_data
+
+    def join_company_metrics(
+        self,
+        metrics: list,
+        years: list,
+        access: bool,
+        companies_by_investment,
+        **conditions,
     ) -> dict:
-        company = dict()
-        metrics_values = list()
+        companies = dict()
         for metric in metrics:
             data = self.get_by_metric_records(
-                metric, years, access, company_id, **conditions
+                metric, years, access, companies_by_investment, **conditions
             )
-            company.update(data[0])
-            metrics_values.append(data[1])
-        company["metrics"] = metrics_values
-        return company
+            company_data = dict()
```

Comment:
Should this company_data be defined in this line?

Reply:
It was deleted.